### PR TITLE
[Yaml] Support tagged scalars

### PR DESCRIPTION
**Pull request by GuilhemN**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes, replacement of deprecations
| BC breaks?    | yes
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        | 

Add support of tagged scalars in 4.0.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
missing CHANGELOG updates in components (please check any your other PRs also if any)

**Comment by GuilhemN** [comment]
@nicolas-grekas note added

**Comment by xabbuh** [comment]
@GuilhemN Can you rebase here now that #22770 is merged?

**Comment by GuilhemN** [comment]
done

File: src/Symfony/Component/Yaml/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,12 @@ CHANGELOG
    the parser and dumper is no longer supported, pass bitmask flags instead
  * the constructor arguments of the `Parser` class have been removed
  * the `Inline` class is internal and no longer part of the BC promise
+ * added support of tagged scalars.
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/CHANGELOG.md]: support for

Comment (GuilhemN) [inline src/Symfony/Component/Yaml/CHANGELOG.md]: updated, thanks

**Comment by fabpot** [comment]
target should be changed to 3.4.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @GuilhemN.

**Pull request merged**